app/src/main/java/com/eveningoutpost/dexdrip/Models/LibreOOPAlgorithm.java

```diff
@@ -311,13 +314,39 @@ public static void handleDecodedBleResult(long timestamp, byte[] ble_data, byte
         glucoseData.glucoseLevelRaw = raw;
         
         libreAlarmObject.data.trend.add(glucoseData);
+
+        //========= add code here for smmothing and gap closing.
```

Comment:
What is the current progress of gap closing?
There is a PR open for smoothing #1401

Reply:
gap closing is implemented in https://github.com/NightscoutFoundation/xDrip/pull/1640
Looking at 1401 to see if I can get ideas from it.